tests/StringyTest.php

```diff
@@ -2482,4 +2482,24 @@ public function htmlDecodeProvider()
             array('>', '&gt;'),
         );
     }
+
+    /**
+     * @dataProvider stripPunctuationProvider()
+     */
+    public function testStripPunctuation($expected, $str, $encoding = null)
+    {
+        $stringy = S::create($str, $encoding);
+        $result = $stringy->stripPunctuation();
+        $this->assertStringy($result);
+        $this->assertEquals($expected, $result);
+        $this->assertEquals($str, $stringy);
+    }
+
+    public function stripPunctuationProvider()
+    {
+        return array(
+            array('foo bar ', '!{foo}:,+"`-=\/ (&bar.) @#'),
+            array('fòô bàř 123', '!{fòô}: (bàř)? №1;2--__3%', 'UTF-8'),
```

Comment:
Isn't this stripping a lot more than just punctuation? https://en.wikipedia.org/wiki/Punctuation#In_computing